### PR TITLE
fv3fit: remove sklearn regressor ensemble

### PR DESCRIPTION
**Pull request by brianhenn**
Removes the fv3fit sklearn's internal `_RegressorEnsemble` class, since its random forests are no longer trained as an ensemble this way.  Backwards compatible loading of `RandomForest` models created after the change to a single sklearn ensemble regressor is allowed, but not before.

Significant internal changes:
- When training `sklearn_random_forest`,  the model wrapper now accepts a sklearn `RandomForestRegressor` as its regressor instead of the `_RegressorEnsemble`. 
- The 'input_sensitivity' method previously on the `SklearnWrapper` is now on the `RandomForest` and accesses the `RandomForestRegressor`'s ensemble sensivities, which restores the ability of the offline diags to show this for the RF.

- [X] Tests added

Coverage reports (updated automatically):
- test_unit: [61%](https://output.circle-artifacts.com/output/job/8419bf9a-8d8e-4283-b392-e35c62501c5e/artifacts/0/tmp/coverage/htmlcov-test_unit/index.html)

### CODE REVIEW

File: external/fv3fit/fv3fit/sklearn/_random_forest.py [hunk 1]
```diff
@@ -356,40 +297,44 @@ def load(cls, path: str) -> "SklearnWrapper":
...
-        return np.array(self.feature_importances).mean(axis=0)
-
-    @property
-    def std_importances(self) -> np.ndarray:
-        return np.array(self.feature_importances).std(axis=0)
+    @staticmethod
+    def _load_regressor(b: bytes) -> Tuple[sklearn.base.BaseEstimator, int]:
+        regressor_components = joblib.load(io.BytesIO(b))
+        if isinstance(regressor_components["regressors"], list):
+            # backward compatibility for previous models saved as single batch regressor
```
Comment (brianhenn) [inline external/fv3fit/fv3fit/sklearn/_random_forest.py:318]: Could add a test of this backwards compatibility.

File: external/fv3fit/fv3fit/sklearn/_random_forest.py [hunk 2]
```diff
@@ -128,84 +125,8 @@ def dump(self, path: str) -> None:
...
-            "regressors"
-        ]
-        base_regressor = batch_regressor_components["base_regressor"]
-        obj = cls(
-            base_regressor=base_regressor,
-            regressors=regressors,
-            n_jobs=batch_regressor_components.get("n_jobs", 1),
-        )
-        return obj
+    def input_sensitivity(self, stacked_sample: xr.Dataset) -> InputSensitivity:
```
[the remark this answers is not in the document]
Comment (AnnaKwa) [reply external/fv3fit/fv3fit/sklearn/_random_forest.py:128]: Do the current offline reports include this for RFs? I haven't looked at this for a RF since before the refactor of the model. If the method isn't implemented the report just doesn't include those plots.

File: external/fv3fit/fv3fit/sklearn/_random_forest.py [hunk 3]
```diff
@@ -356,40 +297,44 @@ def load(cls, path: str) -> "SklearnWrapper":
...
     def input_sensitivity(self, stacked_sample: xr.Dataset) -> InputSensitivity:
         _, input_multiindex = pack(
             stacked_sample[self.input_variables], ["sample"], self.packer_config
         )
         feature_importances = {}
         for (name, feature_index), mean_importance, std_importance in zip(
-            input_multiindex, self.mean_importances, self.std_importances
+            input_multiindex,
+            self.model.feature_importances_,
+            np.zeros_like(self.model.feature_importances_),
```
[the remark this answers is not in the document]
Comment (AnnaKwa) [reply external/fv3fit/fv3fit/sklearn/_random_forest.py:337]: We can still keep the standard deviations in the returned data by adding something like`std_importances = np.std([tree.feature_importances_ for tree in self.model.estimators_], axis=0)`

File: external/fv3fit/fv3fit/sklearn/_random_forest.py [hunk 4]
```diff
@@ -356,40 +297,44 @@ def load(cls, path: str) -> "SklearnWrapper":
...
+        return regressor, regressor_components["n_jobs"]
 
     def input_sensitivity(self, stacked_sample: xr.Dataset) -> InputSensitivity:
         _, input_multiindex = pack(
             stacked_sample[self.input_variables], ["sample"], self.packer_config
         )
         feature_importances = {}
         for (name, feature_index), mean_importance, std_importance in zip(
-            input_multiindex, self.mean_importances, self.std_importances
+            input_multiindex,
```
Comment (AnnaKwa) [inline external/fv3fit/fv3fit/sklearn/_random_forest.py:335]: The rest of the `SklearnWrapper` class now works for generic sklearn estimators, with the exception of this method. Could you make a child `SklearnWrapperRF` class that inherits from `SklearnWrapper` and has this method for itself?

**Review by AnnaKwa — APPROVED** [review]
I have a couple minor requests with regards to the feature sensitivity code but other than that LGTM and I don't think I need to rereview after those changes.

File: external/fv3fit/fv3fit/sklearn/_random_forest.py (hunk 4 above)
Comment (brianhenn) [reply external/fv3fit/fv3fit/sklearn/_random_forest.py:335]: As discussed will move this to a method of the RandomForest rather than of the SklearnWrapper, which should achieve the same goal of having the SklearnWrapper work for any sklearn estimator.

File: external/fv3fit/fv3fit/sklearn/_random_forest.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (brianhenn) [reply external/fv3fit/fv3fit/sklearn/_random_forest.py:128]: They do (see `gs://vcm-ml-scratch/test-end-to-end-integration/2023-02-18/cc57ab7-a764-train-diags-prog/offline-report/sklearn_dQ1_model` for example), which is why I was confused.

[the remark this answers is not in the document]
Comment (brianhenn) [reply external/fv3fit/fv3fit/sklearn/_random_forest.py:128]: Ha never mind, it looks like the link to a report with the sensitivity figure I sent was from this PR's test (which fixed it). It does look like recent e2e tests didn't include that plot for the RF.
